File: twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            map.put(target - nums[i], i);
        }

        for (int i = 0; i < nums.length; i++) {
            Integer targetIndice = map.get(nums[i]);
            if (targetIndice != null) {
                if (i == targetIndice) {
                    continue;
                }
                result[0] = targetIndice < i ? targetIndice : i;
                result[1] = targetIndice < i ? i : targetIndice;
            }
        }

        return result;
    }
}

File: twoSum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    var result = <int>[];
    var map = Map<int, int>();
    for (int i = 0; i < nums.length; i++) {
        map[target - nums[i]] = i;
    }
    for (int i = 0; i < nums.length; i++) {
        int targetNumber = nums[i];
        int? targetIndex = map[targetNumber];
        
        if (targetIndex == i || targetIndex == null) {
            continue;
        }

        if (targetIndex < i) {
            result = [targetIndex, i];
            break;
        }

        result = [i, targetIndex];
    }

    return result;
  }
}
